src/sql/sqlops.proposed.d.ts

```diff
@@ -325,4 +325,28 @@ declare module 'sqlops' {
 			}
 		}
 	}
+
+	/**
+	 * Namespace for interacting with query editor
+	*/
+	export namespace queryEditor {
+		/**
+		 * Create new query editor
+		 * @param {string} queryContent The content of query editor
+		 * @returns {string} The file URI of the new query editor
+		*/
+		export function newQueryEditor(queryContent?: string): Thenable<string>;
```

Comment:
minor comment - might be better to call this `newEditor` and the argument `content` since `queryEditor` is already the namespace